### PR TITLE
BUG: Fixes module data docstrings.

### LINKED ISSUE #15325: BUG: f2py does not report names of module data items
Current output from [this example](https://numpy.org/devdocs/f2py/python-usage.html#allocatable-arrays) in the docs:

```python
>>> import allocarr
>>> print(allocarr.mod.__doc__)
'f'-
foo()

Wrapper for ``foo``.


>>> 
```

Expected output:

```python
>>> import allocarr 
>>> print(allocarr.mod.__doc__)
b - 'f'-array(-1,-1), not allocated
foo - Function signature:
  foo()

>>>
```

f2py output is also incomplete in other situations (see for example #15308 )

```python
>>> import sys, numpy; print(numpy.__version__, sys.version)
1.19.0.dev0+edb5f56 3.6.7 | packaged by conda-forge | (default, Jul  2 2019, 02:18:42) 
[GCC 7.3.0]
```

### PR DESCRIPTION
**Pull request by melissawm**
Improves printing of module data items, showing proper information via the module docstring.

Fixes #15325 and #15308

### CODE REVIEW

File: numpy/f2py/src/fortranobject.c [hunk 1]
```diff
@@ -213,6 +213,7 @@ format_def(char *buf, Py_ssize_t size, FortranDataDef def)
             return -1;
         }
         memcpy(p, notalloc, sizeof(notalloc));
+        p += sizeof(notalloc);
```
Comment (eric-wieser) [inline numpy/f2py/src/fortranobject.c:216]:
```suggestion
        p += sizeof(notalloc);
        size -= sizeof(notalloc);
```
I think is needed

**Comment by melissawm** [comment]
Testing with CI, I noticed 
```
    @pytest.mark.skipif(sys.platform=='win32',
                        reason='Fails with MinGW64 Gfortran (Issue #9673)')
    @pytest.mark.xfail(IS_PYPY,
                       reason="PyPy cannot modify tp_doc after PyType_Ready")
```
were necessary to get the new test to run - but I'm not really sure if those are the right messages, as I'm not familiar with either PyPy or Windows. From #9673 and `test_block_docstring.py` it looks like those are appropriate.

File: numpy/f2py/src/fortranobject.c [hunk 2]
```diff
@@ -264,7 +266,7 @@ fortran_doc(FortranDataDef def)
         size -= n;
 
         if (def.data == NULL) {
-            n = format_def(p, size, def) == -1;
+            n = format_def(p, size, def);// == -1;
```
Comment (charris) [inline numpy/f2py/src/fortranobject.c:269]: Might as well remove the comparison as comment it out.

Comment (melissawm) [reply numpy/f2py/src/fortranobject.c:269]: Sorry, that was supposed to go when I took the PR out of draft. Done!

File: numpy/f2py/tests/test_module_doc.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+class TestModuleDocString(util.F2PyTest):
+    sources = [_path('src', 'module_data', 'module_data_docstring.f90')]
+
+    @pytest.mark.skipif(sys.platform=='win32',
+                        reason='Fails with MinGW64 Gfortran (Issue #9673)')
+    @pytest.mark.xfail(IS_PYPY,
+                       reason="PyPy cannot modify tp_doc after PyType_Ready")
+    def test_module_docstring(self):
+        expected = "i : 'i'-scalar\nx : 'i'-array(4)\na : 'f'-array(2,3)\nb : 'f'-array(-1,-1), not allocated\x00\nfoo()\n\nWrapper for ``foo``.\n\n"
```
Comment (eric-wieser) [inline numpy/f2py/tests/test_module_doc.py:21]: This would be a good use of `textwrap.dedent('"""...""")`

Comment (melissawm) [reply numpy/f2py/tests/test_module_doc.py]: Thanks for the suggestion, I was not familiar with that :)

**Comment by charris** [comment]
Thanks Melissa.

**Pull request merged**